src/frontend/src/handler/create_mv.rs

```diff
@@ -200,35 +236,75 @@ pub async fn handle_create_mv_bound(
     }
 
     let (table, graph, dependencies, resource_group) = {
-        let context = OptimizerContext::from_handler_args(handler_args);
-        let mut with_options = context.with_options().clone();
-
+        let mut with_options = get_with_options(handler_args.clone());
         let resource_group = with_options.remove(&RESOURCE_GROUP_KEY.to_owned());
 
-        if resource_group.is_some()
-            && !context
-                .session_ctx()
-                .config()
-                .streaming_use_arrangement_backfill()
-        {
-            return Err(RwError::from(ProtocolError("The session config arrangement backfill must be enabled to use the resource_group option".to_owned())));
+        let is_serverless_backfill = with_options
+            .remove(&CLOUD_SERVERLESS_BACKFILL_ENABLED.to_owned())
```

Comment:
Remove?

Reply:
We need this. `is_serverless_backfill` is true if the user passed `cloud.serverless_backfill_enabled=true` when creating the materialised view.

Reply:
Didn't get it. Why we need to remove it?

Reply:
We also remove `let resource_group = with_options.remove(&RESOURCE_GROUP_KEY.to_owned());` because we want to check if we processed all `WITH` statements from the query: 

```
if !with_options.is_empty() {
    // get other useful fields by `remove`, the logic here is to reject unknown options.
    return Err(RwError::from(ProtocolError(format!(
        "unexpected options in WITH clause: {:?}",
        with_options.keys()
    ))));
}
```